File: Noble-iOS/Noble-iOS/TacPad View/TacPadViewModel.swift
import shared
import Foundation

class TacPadViewModel: ObservableObject {
    private let audioService: AudioService

    init(audioService: AudioService = AudioService()) {
        self.audioService = audioService
    }

    func playClip(_ soundBite: CarterSoundBites) {
        audioService.playSoundClip(name: soundBite.rawValue)
    }
}

File: Noble-Android/src/main/java/com/fallencosplay/noble_mp/android/tacpad/TacPadViewModel.kt
package com.fallencosplay.noble_mp.android.tacpad

import androidx.lifecycle.ViewModel
import com.fallencosplay.noble_mp.shared.AudioService

class TacPadViewModel(private val audioService: AudioService): ViewModel() {
    fun playClip(name: String) {
        audioService.playSoundClip(name)
    }
}